src/sms.rs

```diff
@@ -0,0 +1,201 @@
+use crate::{
+    error::{Error, ErrorSource},
+    send_at, LteLink,
+};
+
+use core::fmt::Write;
+use core::write;
+use heapless::{String, Vec};
+
+// ASCII table for coverting ASCII to GSM 7 bit
+// Copied from https://github.com/nrfconnect/sdk-nrf/blob/main/lib/sms/string_conversion.c#L36
+
+const ASCII_TO_7BIT_TABLE: [u8; 256] = [
```

Comment:
Because this is const and not static, this should not show up in the final binary if it isn't used. So that's one less reason to put sms behind a feature gate.